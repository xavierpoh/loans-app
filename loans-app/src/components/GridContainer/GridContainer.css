.details-grid {
  display: flex;
  justify-content: center;
  align-items: center;
}

@media (max-width: 600px) {
  .details-grid {
    flex-direction: column;
  }
}