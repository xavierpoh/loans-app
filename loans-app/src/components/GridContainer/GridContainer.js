import React from 'react';
import Grid from './../Grid/Grid';
import './GridContainer.css';

const GridContainer = props => (
  <div className="details-grid">
    {props.headers.map((header, index) => (
        <Grid
          hasMargin={props.headers.length - 1 === index ? false : true}
          header={header} 
          gridData={props.gridData[index]}
          key={index} />
      )
    )}
  </div>
)

export default GridContainer;