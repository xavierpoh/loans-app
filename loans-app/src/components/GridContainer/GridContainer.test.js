import GridContainer from './GridContainer';
import React from 'react';
import { shallow } from 'enzyme';

const Headers = ['Loan Amount', 'Loan Balance', 'Weekly Repayment', 'Remaining Term (Weeks)'];
const GridData = [
  {data: '120', is_money: true}, 
  {data: '100', is_money: true}, 
  {data: '20', is_money: true}, 
  {data: '5', is_money: false}
];

describe('GridContainer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GridContainer headers={Headers} gridData={GridData} />);
    expect(wrapper).toMatchSnapshot();
  })
})