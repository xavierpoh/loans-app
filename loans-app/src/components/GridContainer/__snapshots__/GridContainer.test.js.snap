// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridContainer
    gridData={
        Array [
            Object {
              "data": "120",
              "is_money": true,
            },
            Object {
              "data": "100",
              "is_money": true,
            },
            Object {
              "data": "20",
              "is_money": true,
            },
            Object {
              "data": "5",
              "is_money": false,
            },
          ]
    }
    headers={
        Array [
            "Loan Amount",
            "Loan Balance",
            "Weekly Repayment",
            "Remaining Term (Weeks)",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Grid
          gridData={
                    Object {
                              "data": "120",
                              "is_money": true,
                            }
          }
          hasMargin={true}
          header="Loan Amount"
/>,
        <Grid
          gridData={
                    Object {
                              "data": "100",
                              "is_money": true,
                            }
          }
          hasMargin={true}
          header="Loan Balance"
/>,
        <Grid
          gridData={
                    Object {
                              "data": "20",
                              "is_money": true,
                            }
          }
          hasMargin={true}
          header="Weekly Repayment"
/>,
        <Grid
          gridData={
                    Object {
                              "data": "5",
                              "is_money": false,
                            }
          }
          hasMargin={false}
          header="Remaining Term (Weeks)"
/>,
      ],
      "className": "details-grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "gridData": Object {
            "data": "120",
            "is_money": true,
          },
          "hasMargin": true,
          "header": "Loan Amount",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "gridData": Object {
            "data": "100",
            "is_money": true,
          },
          "hasMargin": true,
          "header": "Loan Balance",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "gridData": Object {
            "data": "20",
            "is_money": true,
          },
          "hasMargin": true,
          "header": "Weekly Repayment",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "gridData": Object {
            "data": "5",
            "is_money": false,
          },
          "hasMargin": false,
          "header": "Remaining Term (Weeks)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Grid
            gridData={
                        Object {
                                    "data": "120",
                                    "is_money": true,
                                  }
            }
            hasMargin={true}
            header="Loan Amount"
/>,
          <Grid
            gridData={
                        Object {
                                    "data": "100",
                                    "is_money": true,
                                  }
            }
            hasMargin={true}
            header="Loan Balance"
/>,
          <Grid
            gridData={
                        Object {
                                    "data": "20",
                                    "is_money": true,
                                  }
            }
            hasMargin={true}
            header="Weekly Repayment"
/>,
          <Grid
            gridData={
                        Object {
                                    "data": "5",
                                    "is_money": false,
                                  }
            }
            hasMargin={false}
            header="Remaining Term (Weeks)"
/>,
        ],
        "className": "details-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "gridData": Object {
              "data": "120",
              "is_money": true,
            },
            "hasMargin": true,
            "header": "Loan Amount",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "gridData": Object {
              "data": "100",
              "is_money": true,
            },
            "hasMargin": true,
            "header": "Loan Balance",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "gridData": Object {
              "data": "20",
              "is_money": true,
            },
            "hasMargin": true,
            "header": "Weekly Repayment",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "gridData": Object {
              "data": "5",
              "is_money": false,
            },
            "hasMargin": false,
            "header": "Remaining Term (Weeks)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
